node/trace/span.js

```diff
@@ -0,0 +1,203 @@
+var Ready = require('ready-signal');
+var crypto = require('crypto');
+var NullLogtron = require('null-logtron');
+
+module.exports = Span;
+module.exports.Endpoint = Endpoint;
+module.exports.Annotation = Annotation;
+module.exports.BinaryAnnotation = BinaryAnnotation;
+
+function mathRng(cb) {
+    cb(null, Math.random() * 100000000000000000);
+}
+
+function cryptoRng(cb) {
```

Comment:
Do we actually need cryptographically secure randomness here? It doesn't seem to me like we should...

If you just need 8 bytes of random, then you could use a simple Linear Congruential Generator, which is fast to compute and not async... I'm adding one to generate test data for streaming: https://github.com/uber/tchannel/commit/7b76ccb67ded605dad2384323ccd6e586d7403f2 , but you don't even need all the overhead of being a stream... just a function that returns 8 bytes will do, which can probably be done by choosing some decent 32-bit constants from http://en.wikipedia.org/wiki/Linear_congruential_generator and then turning the crank twice (fits inside JS's double-math better).